### ISSUE TITLE
ConcurrentModificationException when executing AutowireCapableBeanFactory.createBean

### ISSUE THREAD

**Issue by syedkither** [spring-projects/spring-framework#24561]
We are facing Initialization of bean failed; nested exception is java.util.ConcurrentModificationException and we are using Akka framework 2.6.3, Apache AVRO-1.9.2 and these are in classpath spring-boot-2.0.1.release and spring framework (bean,context,core,web) 5.0.5 release

 

below are the error stack. please help me what needs to be done to resolve this issue

 
```
 akka.actor.ActorInitializationException: exception during creation

	at akka.actor.ActorInitializationException$.apply(Actor.scala:166)

	at akka.actor.ActorCell.create(ActorCell.scala:596)

	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:456)

	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:478)

	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:263)

	at akka.dispatch.Mailbox.run(Mailbox.scala:219)

	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)

	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)

	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)

	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)

	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

        Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with 
        name 'messageHandlerActor' defined in file [/Users/muv416/intellij_workplace/decision- 
        engine/stellar/target/classes/com/fs/decisionengine/stellar/actor/MessageHandlerActor.class]: 
        Initialization of bean failed; nested exception is java.util.ConcurrentModificationException

	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:587)

	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:501)

	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335)

	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)

	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1089)

	at com.capitalone.fs.decisionengine.stellar.common.SpringActorProducer.produce(SpringActorProducer.java:26)

	at akka.actor.Props.newActor(Props.scala:255)

	at akka.actor.ActorCell.newActor(ActorCell.scala:552)

	at akka.actor.ActorCell.create(ActorCell.scala:578)

	... 9 common frames omitted

        Caused by: java.util.ConcurrentModificationException: null

	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)

	at java.util.ArrayList$Itr.next(ArrayList.java:859)

	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1341)

	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:578)

	... 17 common frames omitted
```

**Comment by syedkither** [comment]
@jhoeller please advise me on  how to resolve it ..

**Comment by jhoeller** [comment]
This looks like a duplicate of #21819, fixed in 5.0.10 and higher.
